azurelinuxagent/ga/exthandlers.py

```
@@ -548,108 +561,173 @@ def _report_error_event_and_return_false(error_message):
                 ext_handler_i.set_operation(WALAEventOperation.Download)
-                ext_handler_i.set_handler_status(message=ustr(err_msg), code=-1)
-                ext_handler_i.report_event(message=ustr(err_msg), is_success=False)
-                return False
-
-            ext_handler_i.logger.info("Target handler state: {0} [incarnation {1}]", state, etag)
-            if state == ExtensionRequestedState.Enabled:
-                self.handle_enable(ext_handler_i)
-            elif state == ExtensionRequestedState.Disabled:
-                self.handle_disable(ext_handler_i)
-            elif state == ExtensionRequestedState.Uninstall:
-                self.handle_uninstall(ext_handler_i)
+                raise ExtensionError(msg=err_msg)
+
+            # Handle everything on an extension level rather than Handler level
+            ext_handler_i.logger.info("Target handler state: {0} [incarnation {1}]", handler_state, etag)
+            if handler_state == ExtHandlerRequestedState.Enabled:
+                self.handle_enable(ext_handler_i, extension)
+            elif handler_state == ExtHandlerRequestedState.Disabled:
+                self.handle_disable(ext_handler_i, extension)
+            elif handler_state == ExtHandlerRequestedState.Uninstall:
+                self.handle_uninstall(ext_handler_i, extension=extension)
             else:
-                message = u"Unknown ext handler state:{0}".format(state)
+                message = u"Unknown ext handler state:{0}".format(handler_state)
                 raise ExtensionError(message)
 
             return True
-
+        except MultiConfigExtensionError as error:
+            ext_name = ext_handler_i.get_extension_full_name(extension)
+            err_msg = "Error processing MultiConfig extension {0}: {1}".format(ext_name, ustr(error))
+            # This error is only thrown for enable operation on MultiConfig extension.
```

Should we rename it to MultiConfigExtensionEnableError to make this clearer and help avoid raising those on other operations?

---

Makes sense. Initially I was raising this for both Disable and Enable errors but then added the code to ignore errors for disable, but renaming it makes sense. Let me do it